tests/resolver/test_filtering.py

```diff
@@ -0,0 +1,405 @@
+from __future__ import annotations
+
+import copy
+import random
+from collections import deque
+
+import pytest
+
+from variantlib.errors import ValidationError
+from variantlib.models.variant import VariantDescription
+from variantlib.models.variant import VariantFeature
+from variantlib.models.variant import VariantProperty
+from variantlib.resolver.filtering import filter_variants_by_features
+from variantlib.resolver.filtering import filter_variants_by_namespaces
+from variantlib.resolver.filtering import filter_variants_by_property
+from variantlib.resolver.filtering import remove_duplicates
+
+
+@pytest.fixture(scope="session")
+def vprops() -> list[VariantProperty]:
+    return [
+        VariantProperty(namespace="OmniCorp", feature="custom_feat", value="value1"),
+        VariantProperty(
+            namespace="TyrellCorporation", feature="client_id", value="value2"
+        ),
+    ]
+
+
+@pytest.fixture(scope="session")
+def vdescs(vprops: list[VariantProperty]) -> list[VariantDescription]:
+    """Fixture to create a list of VariantDescription objects."""
+    assert len(vprops) == 2
+    prop1, prop2 = vprops
+
+    return [
+        VariantDescription([prop1]),
+        VariantDescription([prop2]),
+        VariantDescription([prop1, prop2]),
+    ]
```

Comment:
Perhaps make it more universal:

```suggestion
    return list(itertools.chain.from_iterable(itertools.combinations(vprops, x + 1) for x in range(len(vprops))))
```

Reply:
It's equivalent but a lot less readable and hard to understand in just a quick look.
I will not merge this - I prefer keeping the readability a tad better at the expense of a little more code.